### PR TITLE
#1145 Have a way to differentiate UPDATE_PROPERTY propertyListener fr…

### PR DESCRIPTION
**Pull request by nabilIbm**
…om initial load and when user does change

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/properties-controller.js [hunk 1]
```diff
@@ -396,7 +396,7 @@ export default class PropertiesController {
 					controlValue = PropertyUtils.convertObjectStructureToArray(control.valueDef.isList, control.subControls, controlValue);
 				}
 
-				this.updatePropertyValue(propertyId, controlValue, true);
+				this.updatePropertyValue(propertyId, controlValue, true, "initial_load");
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/src/common-properties/properties-controller.js:399]: `_addToControlValues()` method is called during init only. Everywhere you're calling `this.updatePropertyValue()` from this method, you should pass `"initial_load"`.

**Comment by nmgokhale** [comment]
For `oneofselect` control, `type: undefined` during initial load. Seen this in some other paramDefs as well. 
Screen Shot 2022-08-11 at 1 04 34 PM

Please verify `type: initial_load` after loading all paramDefs in test harness.

**Comment by nabilIbm** [comment]
> For `oneofselect` control, `type: undefined` during initial load. Seen this in some other paramDefs as well. Screen Shot 2022-08-11 at 1 04 34 PM
> 
> Please verify `type: initial_load` after loading all paramDefs in test harness.

image


> For `oneofselect` control, `type: undefined` during initial load. Seen this in some other paramDefs as well. Screen Shot 2022-08-11 at 1 04 34 PM
> 
> Please verify `type: initial_load` after loading all paramDefs in test harness.

image

**Comment by nabilIbm** [comment]
> For `oneofselect` control, `type: undefined` during initial load. Seen this in some other paramDefs as well. Screen Shot 2022-08-11 at 1 04 34 PM
> 
> Please verify `type: initial_load` after loading all paramDefs in test harness.

image

**Comment by nabilIbm** [comment]
Closed

**Pull request closed without merging**